docs/icicle/primitives/msm.md

```diff
@@ -27,18 +96,23 @@ msm::msm(&scalars, &points, &cfg, &mut msm_results).unwrap();
 ...
 ```
 
-`is_big_triangle=true` will activate Large triangle accumulation and `is_big_triangle=false` will activate Bucket accumulation.
-
-#### MSM Modes
+You may reference the rust code [here](https://github.com/ingonyama-zk/icicle/blob/77a7613aa21961030e4e12bf1c9a78a2dadb2518/wrappers/rust/icicle-core/src/msm/mod.rs#L54).
 
-ICICLE MSM also supports two different modes
+## MSM Modes
 
-- Batch MSM
-- Single MSM
+ICICLE MSM also supports two different modes `Batch MSM` and `Single MSM`
 
 Batch MSM allows you to run many MSMs with a single API call, Single MSM will launch a single MSM computation.
 
-You don't need to do anything special to toggle between batch or single MSM.
+### Which mode should I use ?
+
+This decision is highly dependent on your use case and design. However if your design allows for it using batch mode can significantly improve efficiency. Batch processing allows you to perform multiple MSM leveraging parallel processing capabilities of the GPU.
+
+Single MSM mode should be used when batching isn't possible or when you have to run a single MSM.
+
+### How do I toggle between MSM modes?
+
+You simply call `msm::msm`, if `msm_results` is a single result it will run in single MSM mode, if you are expecting many results MSM will run in batch mode, using the number of expected results in `msm_results` as the `batch_size`, the scalars and points will be split into groups based on `batch_size` and processed in parallel.
```

Comment:
This should be reworded. Not sure exactly how but it doesn't read well

Reply:
@jeremyfelder please tell me what you think now.